### PR TITLE
applayer: add tests for bug 7044

### PR DESCRIPTION
**Pull request by inashivb**
## Ticket

If your pull request is related to a Suricata ticket, please provide
the full URL to the ticket here so this pull request can monitor
changes to the ticket status:

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/7044

Previous PR: https://github.com/OISF/suricata-verify/pull/1946

Changes since v2:
- a min-version check was added
Reason: Due to `truncate` callback in 6, a `REQUEST` was logged as a local variable `req_done` was set to `true`, however, the request logged in such a case was empty. So, it isn't much different from the behavior in `master` or `7.0.x` where the request is logged as "lost" bc the request never had anything and there was no callback to make it available for logging.

### CODE REVIEW

File: tests/truncate-applayer-test-01/test.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+requires:
+  min-version: 7
+
+checks:
+- filter:
+    count: 1
+    match:
+      event_type: alert
+      alert.signature_id: 1
```
Comment (catenacyber) [inline tests/truncate-applayer-test-01/test.yaml:12]: maybe add pcap_cnt ?

File: tests/truncate-applayer-test-02/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Test Description
+
+This test demonstrates that the pcap is properly parsed and assessed in case
+there was no stream reassembly depth constraints.
```
Comment (catenacyber) [inline tests/truncate-applayer-test-02/README.md:4]: there is the default one ;-)

**Review by catenacyber — APPROVED** [review]
Details can be added, but looks good to me to check current behavior and that truncate removal does not affect it

File: tests/truncate-applayer-test-02/README.md (hunk 2 above)
Comment (inashivb) [reply tests/truncate-applayer-test-02/README.md:4]: indeed. Wil change the words. Thanks :)

**Pull request closed without merging**